src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const ch of actions) {
+    const clonedState = { ...currentState };
+
+    if (ch.type === 'addProperties') {
+      for (const [key, value] of Object.entries(ch.extraData)) {
+        clonedState[key] = value;
+      }
+    }
+
+    if (ch.type === 'removeProperties') {
+      for (const key of ch.keysToRemove) {
+        delete clonedState[key];
+      }
+    }
+
+    if (ch.type === 'clear') {
+      for (const key in clonedState) {
+        delete clonedState[key];
+      }
+    }
+
+    result.push(clonedState);
+    currentState = clonedState;
```

Comment:
Instead of using multiple if statements, consider refactoring the code to use a switch statement, as you have a limited amount of conditions to check. This will make your code cleaner and more maintainable. [CODE STYLE]: Use switch statement if you have limited amount of conditions.